### PR TITLE
offlinePlayerScoreAPI

### PR DESCRIPTION
**Pull request by kkj9333**
## Description

Please carefully read the [Contributing note](https://docs.litebds.com/#/zh_CN/Maintenance/README) before making any pull requests.
And, **Do not make a pull request to merge into main branch unless it is a hotfix. Use the beta branch instead.**
## Type of changes

- [ ] Bug fix
- [ ] New feature 
- [x] Enhancement
- [ ] Documentation

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://docs.litebds.com/#/zh_CN/Maintenance/README)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

**Review by ShrBox — APPROVED** [review]

File: LiteLoader/include/llapi/mc/Scoreboard.hpp [hunk 1]
```diff
@@ -122,6 +182,15 @@ class Scoreboard {
     LIAPI static bool deleteScore(Player* player, const std::string& objname);
 
     LIAPI static bool scoreboardIdIsValid(ScoreboardId* id);
+
+     /**
+     * @brief return new available ScoreboardId, don't use directly
+     * 
+     * @return ScoreboardId const reference 
+     */
+
```
Comment (Jaciezyt) [inline LiteLoader/include/llapi/mc/Scoreboard.hpp:191]: ？

File: LiteLoader/include/llapi/mc/PlayerScoreboardId.hpp [hunk 2]
```diff
@@ -20,7 +20,29 @@ struct PlayerScoreboardId {
...
 // Add Member There
+    //acotid
+    long long Actoruniqueid;
+    PlayerScoreboardId():id(-1)
+    {
 
+    }
+
+    PlayerScoreboardId(ActorUniqueID const& aid):id(aid.id)
+    {
```
Comment (Jaciezyt) [inline LiteLoader/include/llapi/mc/PlayerScoreboardId.hpp:31]: clang-format plz

File: LiteLoader/include/llapi/mc/Scoreboard.hpp [hunk 3]
```diff
@@ -36,8 +36,68 @@ class Scoreboard {
...
+    }
 public:
 
+    /**
+    * @brief create a new available ScoreboardId for Player by mce::UUID
+    * 
+    * The user should check uuid's validity
+    * @return bool
+    */
+    LIAPI bool createPScoreBoardId(mce::UUID const& uuid);
```
Comment (Jaciezyt) [inline LiteLoader/include/llapi/mc/Scoreboard.hpp:70]: 不符合命名规范，请使用全称，此外，ScoreboardId作为一个类名，不应修改其大小写

File: LiteLoader/include/llapi/mc/Scoreboard.hpp [hunk 4]
```diff
@@ -36,8 +36,68 @@ class Scoreboard {
...
+    /**
+    * @brief modify Score for Player by xuid forcely
+    * 
+    * Create forcibly if the objective does not exist
+    * If the player exists but the ScoreboardId does not exist, create it forcibly
+    * modify scores their scores whether they are online or not
+    * return false only when xuid is wrong or runtime-error
+    * @return bool
+    */
+    LIAPI static bool forceModifyPlayerScore(xuid_t const& xuid, std::string const& objname,int val, PlayerScoreSetFunction pf);
```
Comment (Jaciezyt) [inline LiteLoader/include/llapi/mc/Scoreboard.hpp:90]: clang-format plz

**Review by Jaciezyt — CHANGES_REQUESTED** [review]
编译未通过，并存在一些小问题

File: LiteLoader/include/llapi/mc/Scoreboard.hpp [hunk 5]
```diff
@@ -36,8 +36,68 @@ class Scoreboard {
...
+    * The user should check PlayerScoreBoardId is valid or not
+    * @return bool
+    */
+    template <typename T>
+    inline bool createScoreboardId(T const&) = delete;
+    template<>
+    inline bool createScoreboardId(PlayerScoreboardId const& pid) {
+        ScoreboardId newsid = nextScoreboardId();
+        ScoreboardId* (Scoreboard:: * rv)(ScoreboardId, PlayerScoreboardId const&) = NULL;
+        *((void**)&rv) = SYM("??$_createScoreboardId@UPlayerScoreboardId@@@Scoreboard@@IEAAAEBUScoreboardId@@U1@AEBUPlayerScoreboardId@@@Z");
```
Comment (Jaciezyt) [inline LiteLoader/include/llapi/mc/Scoreboard.hpp:59]: bdx时代写法，不推荐

**Review by Jaciezyt — CHANGES_REQUESTED** [review]

File: LiteLoader/include/llapi/mc/Scoreboard.hpp (hunk 1 above)
Comment (kkj9333) [reply LiteLoader/include/llapi/mc/Scoreboard.hpp:191]: 后面想了一下就没必要常引用

File: LiteLoader/include/llapi/mc/PlayerScoreboardId.hpp (hunk 2 above)
Comment (kkj9333) [reply LiteLoader/include/llapi/mc/PlayerScoreboardId.hpp:31]: 我的IDE一开clang-format就一直报错..复制粘贴都报错

File: LiteLoader/include/llapi/mc/Scoreboard.hpp (hunk 5 above)
Comment (kkj9333) [reply LiteLoader/include/llapi/mc/Scoreboard.hpp:59]: LL导出的api里没有这个符号的，所以只能手动sym了

**Pull request closed without merging**